### ISSUE TITLE
fix: TypeError: Promise resolver undefined is not a function

### ISSUE THREAD

**Issue by moltar** [typestack/class-transformer#261]
Not sure if this is an error, or expected behaviour.

```ts
import { plainToClass } from 'class-transformer'

class Foo {
  public num: number

  public str: Promise<string>
}

describe('Foo', () => {
  it('should turn to object', () => {
    const res = plainToClass(Foo, {
      num: 1,
      str: Promise.resolve('a'),
    })

    expect(res.num).toBe(1)
    expect(res.str).resolves.toBe('a')
  })
})
```

This fails with:

```
TypeError: Promise resolver undefined is not a function
    at new Promise (<anonymous>)

  at TransformOperationExecutor.Object.<anonymous>.TransformOperationExecutor.transform (src/TransformOperationExecutor.ts:123:32)
  at _loop_1 (src/TransformOperationExecutor.ts:246:43)
  at TransformOperationExecutor.Object.<anonymous>.TransformOperationExecutor.transform (node_modules/class-transformer/TransformOperationExecutor.js:260:17)
  at ClassTransformer.Object.<anonymous>.ClassTransformer.plainToClass (src/ClassTransformer.ts:43:25)
  at Object.plainToClass (src/index.ts:37:29)
  at Object.it (__tests__/extractors/transform.test.ts:11:17)
```

Any ideas why?

Thanks!

**Comment by PachowStudios** [comment]
Also running into this issue when using graphql-upload with NestJS and the ValidationPipe.

**Comment by lookapanda** [comment]
> Also running into this issue when using graphql-upload with NestJS and the ValidationPipe.

Any solution?

**Comment by PachowStudios** [comment]
@lookapanda Not really a solution but my workaround was to `@Exclude()` the upload field.

**Comment by jbbr** [comment]
@PachowStudios thanks, this prevents the error - but if the field is excluded, it's also not available when using the class/dto.
Have you found any solution for this?

Unfortunately also `@Transform`-decorator is not usable as workaround for this issue as its executed after & additionally to default transform.

**Comment by PachowStudios** [comment]
@jbbr We pass `transform: false` to `ValidationPipe` so you just get the raw DTO. This of course prevents you from doing any kind of transformation, however this has worked fined for us so far.

Another option could be to create your own `ValidationPipe` (which we also did, but for unrelated reasons), and have it handle transforming the Promise fields.

**Side note**: The reason we made our own `ValidationPipe` was because the stock NestJS one recursively strips prototype keys from the input. We use sentry.io which apparently tacks a context property on to `Promise` which contains an infinite loop. As a result, we had to make our own custom pipe which doesn't have this behavior. Hopefully this helps anyone running into something similar.

Regardless, this issue needs to be fixed.
